scripts/ci/build.sh

```diff
@@ -2,17 +2,24 @@
 set -ex
 
 ROOTDIR="$(dirname "$0")/../.."
+PRE_RELEASE="$1"
+
 cd "${ROOTDIR}"
 
 # shellcheck disable=SC2166
 if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]
 then
     echo -n "" >prerelease.txt
 else
-    # Use last commit date rather than build date to avoid ending up with builds for
-    # different platforms having different version strings (and therefore producing different bytecode)
-    # if the CI is triggered just before midnight.
-    TZ=UTC git show --quiet --date="format-local:%Y.%-m.%-d" --format="ci.%cd" >prerelease.txt
+    # Use the given prerelease string if provided
+    if [[ "$PRE_RELEASE" =~ ^[a-zA-Z]+\.[0-9]{4}\.[0-9]{1,2}\.[0-9]{1,2}$ ]]; then
+        echo "$PRE_RELEASE" >prerelease.txt
```

Comment:
Why the regex? Having the script silently ignore the value you explicitly passed in a parameter does not seem like a good idea.

Reply:
Yeah, my idea was to validate the prerelease when provided, but indeed silently ignoring it when it is invalid may lead to unexpected versioning. I will change it to fail in case a invalid prerelease version is provided.

Reply:
This comment was resolved but I see that the regex is still here. Same with my other comment below. I think you applied changes only to `ci/build_emscripten.sh` and must have forgotten about this file.

Reply:
Removed.